### PR TITLE
Don't check against product name if checking server version

### LINKED ISSUE #3001: The server version 21.0.0.18-Nextcloud is unsupported! Proceed at your own risk.
* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
No error about an unsuportes server version should be shown as these version is the official stable one.

### Actual behaviour
Client shows an error message:
_The server version 21.0.0.18-Nextcloud is unsupported! Proceed at your own risk._

### Steps to reproduce
I'm running two instances of nextcloud. Error message is shown on borth. So the steps should be
1. install nextcloud server (stable) 21.0.0.18
2. install client version 3.2 RC1
https://help.nextcloud.com/t/desktop-client-3-2rc1-ready-help-test/110579
3. Connect to server and use virtual file system

### Client configuration
Client version:
- 3.2 RC1
- using virtual file system

Operating system:
- Windows 10, Version 21H1 (Build 19043.867)

OS language:
- German

Installation path of client:
- default location

### Server configuration

Nextcloud version:
- 21.0.0.18

Storage backend (external storage):
- local file system

### PR DESCRIPTION
**Pull request by FlexW**
We should not rely on the product name because it is something the
user can change in the theming options on the server.

Fixes #3001

Signed-off-by: Felix Weilbach <felix.weilbach@nextcloud.com>

### CODE REVIEW

File: version.h.in [hunk 1]
```diff
@@ -19,6 +19,8 @@
 #ifndef VERSION_H
 #define VERSION_H
 
+#include <cstdint>
```
Comment (allexzander) [inline version.h.in:22]: @FlexW Just wondering. Why is this needed?

Comment (FlexW) [reply version.h.in:22]: Ah it's not needed anymore. I originally used uint32_t instead of int. But I then decided to stick with int. I remove it :)

**Review by allexzander — APPROVED** [review]

**Comment by er-vin** [comment]
For the record, I think this came originally from an attempt on making sure nextcloud client is not used on an owncloud server and vice versa. So you might want to come up with some alternative there.

Yes people do try this and then complain when it breaks, hence why it's interesting to catch it early and warn them so that they know what awaits. :-)

**Comment by FlexW** [comment]
@er-vin I checked that with the server guys and they said that we should not rely on product name for that. They said that I should use editon name for that but editon name appears to be empty...

**Comment by er-vin** [comment]
Isn't it something like:
 * unbranded, productname says "Nextcloud" + edition is empty
 * branded, productname says whatever the brand name is + edition says something like Nextcloud

In any case, something is definitely needed. This will come back to haunt you otherwise.

**Comment by FlexW** [comment]
The problem is you can on your Nextcloud server go to Settings -> Theming -> Name and change the name there. The name chosen there will be the name that appears as product name. I see the problem. I will talk to the server people. Maybe they can tell me a reliable way to detect if it's a ownCloud or Nextcloud server.

**Comment by allexzander** [comment]
@FlexW @er-vin This can be a dumb approach, but, we could just check if there is "ownCloud" as a product name, and then, assume it is unsupported :) Of course, this doesn't save us from what @FlexW suggested regarding the Settings, but, still, should work in most cases.

**Comment by er-vin** [comment]
> This can be a dumb approach, but, we could just check if there is "ownCloud" as a product name, and then, assume it is unsupported :)

Thought about that... and then realized they have their own branding as well and likely similar theming in the settings. At least that'd catch the unbranded ones of course, that's already something.
